app/lib/url.rb

```diff
@@ -7,7 +7,7 @@ def self.protocol
   end
 
   def self.domain
-    ApplicationConfig["APP_DOMAIN"]
+    Rails.application&.initialized? ? SiteConfig.app_domain : ApplicationConfig["APP_DOMAIN"]
```

Comment:
I'm not sure how I feel about this. Maybe things that rely on the environment variable before Rails is fully initialized should use it directly instead of through the `URL` module?

Reply:
I ran into some less-important situations right now that technically include this before boot, but in unimportant ways which could change so I think we can keep this and circle back to it. All current forems will have the same value for env and site config stuff right now.